test/Worker.Extensions.Tests/Cosmos/CosmosDBConverterTests.cs

```diff
@@ -384,6 +389,32 @@ public async Task ConvertAsync_POCO_IdProvided_StatusNot200_ThrowsException_Retu
             Assert.Equal("test failure", conversionResult.Error.Message);
         }
 
+        [Fact]
+        public async Task ConvertAsync_POCO_IdProvided_Status404_ThrowsException_ReturnsFailure()
```

Comment:
This should be updated to match the test expectations and outcome:

```suggestion
        public async Task ConvertAsync_POCO_IdProvided_Status404_ReturnsSuccess()
```